### ISSUE TITLE
Uniqueness key for array comparison

### ISSUE THREAD

**Issue by Halynsky** [ltwlf/json-diff-ts#103]
The goal is to compare 2 arrays. As I see in the source, this library should be capable to do that. But there is an issue with key of elements.
For example we have 2 arrays and we want to specify that uniqueness key is - **id**. But in the options we can specify only **embeddedObjKeys**.
I am not sure that it's by design but I found such kind of workaround with empty key of embeddedObjKeys dictionary:

```ts
    console.log(diff(
      [
        {id: 1, name:"obj1"},
        {id: 2, name:"obj2"},
        {id: 3, name:"obj3"}
      ],
      [
        {id: 1, name:"obj1"},
        {id: 3, name:"objxxx"},
        {id: 4, name:"obj4"}
      ],
      {"": "id"}
    ))
```

But the issue is that it inner arrays becomes also be tracked by id. 
So this code will not give any changes.

```ts
    console.log(diff(
      [
        {id: 1, name:"obj1", children: [{name: "child1"}, {name: "child2"}]},
      ],
      [
        {id: 1, name:"obj1", children: [{name: "child2"}]},
      ],
      {"": "id"}
    ))
```

So if we add id for children. It works more correctly and show the changes. 
(But for example I have a case where children don't have an id)

```ts
    console.log(diff(
      [
        {id: 1, name:"obj1", children: [{id: 1, name: "child1"}, {id: 2, name: "child2"}]},
      ],
      [
        {id: 1, name:"obj1", children: [{id: 2, name: "child2"}]},
      ],
      {"": "id"}
    ))
```
Another option that also works is to specify children key. I think it kind of override. - `{"": "id", "children": "name"}`. But this is not very good for big and complicated objects with many child arrays.

Suggestions:
Would be nice to have a way to specify just a key for array. And this key should be not used by child array.
Way how it specified now is ok but at least need to be documented. Alternatives: {".": "id"} or {"$root": "id"}

**Comment by ltwlf** [comment]
In the version 1.2.6 of the software, all paths are treated as regular expressions, this was an issue. However, this approach has been revised in version 2 (ECMAScript Module). In consider for a future update to consider  "" and "." to consider as root. Currently onlöy "" works as root.

In the context of how it's supposed to work, the notation `{"": "id", "children": "name"}` is correct. However, in version 1.2.6, the path `children` would be interpreted as `/children/`, matching any path containing the word "children". To precisely match the 'children' property of the root in version 1.2.6, it's recommended to use the regular expression `^children$`.

Additionally, both versions allow the specification of a function to resolve the key for comparsion  and name of the key, as detailed in the documentation of version 2. This functionality is also present in version 1.2.6.

**Comment by ltwlf** [comment]
Fixes in  #108
